### DISCUSSION TITLE
How to open a file on a single click?

### DISCUSSION THREAD

**Discussion by juli1** [zed-industries/zed#10989] [Feature Requests]
When I click on a file in the file tab, the file open but the cursor/focus is not on the editor. I need to double click on the file to have focus on the editor.

The main issue comes from the confusion in terms of UX: when I click once, the code appears in the editor but I cannot move the cursor. I need to double click to bring the cursor. If the editor shows the code, I expect to bring the focus (e.g. as IntelliJ does: a single click selects the file but does not show it to the user, a double click shows the file and brings focus).

Is there a way (config file) to open the editor and focus the cursor on a single click?

Additional question: what is the binding to switch from the editor to the file panel?

**Comment by nstlgy** [comment]
This feature was recently added to zed as, preview tabs where on single click to view file and to pin them you have to double click the file, #4922 

you can put this in your `settings.json` to turn this feature off 

```  
"preview_tabs": {
    // Whether preview tabs should be enabled.
    // Preview tabs allow you to open files in preview mode, where they close automatically
    // when you switch to another file unless you explicitly pin them.
    // This is useful for quickly viewing files without cluttering your workspace.
    
    "enabled": false,
    
    // Whether to open files in preview mode when selected from the file finder.
    
    "enable_preview_from_file_finder": false
  },
  ```
  
  
 Also, the binding to switch from editor to explorer is `CMD-SHIFT-E`

**Reply by juli1** [reply]
Thank you for the binding! For the configuration, it still does not switch to the editor when opening the file. Is there a way to configure zed to automatically open the file when selecting it?

Thanks!

**Reply by bennetbo** [reply]
As of right now there is no way to change the behavior for a single click, it will always keep the project panel focused. 

Here are a few alternatives which might be helpful to you:
- You can use `project_panel::ToggleFocus` to move the focus to the editor (by default bound to `cmd-shift-e`) after single clicking on a file
- Rebind `project_panel::Open` to `project_panel::OpenPermanent`, allowing you to automatically focus the editor when opening the selected item inside the project panel via `space`

```json
[
  // other keybinds
  {
    "context": "ProjectPanel && not_editing",
    "bindings": {
      "space": "project_panel::OpenPermanent",
    }
  },
  // other keybinds
]
```

**Comment by douglance** [comment]
It would be nice to open the file when clicking on the filename when Zed is *not* focused.

Right now, it just highlights the file and focuses Zed.
